### ISSUE TITLE
add option to "allow in container"

### ISSUE THREAD

**Issue by pszypowicz** [mozilla/multi-account-containers#2222]
### Actual behavior

I want to use seperate container for mail.google.com and calendar.google.com. I want to use option: "Limit to designated Sites".

Unfortunately as soon as login redirects to account.google.com you are forced out of container. 

### Expected behavior

Containers should have 2 lists:

- always open in container
- allow to open in container

So to the each container I can now add account.google.com and it will stay in the same container for login purpose.

**Comment by lcgogo** [comment]
have the same feature request.

**Comment by venteto** [comment]
### TLDR
I basically really need these _overlapping_, _interrelated_ features:

- the ability _**allow**_/whitelist domains in a container
- the ability to then _**restrict**_ the container (and any other container) to only those domains
- the ability to _**not be required**_ to always open those domains in those containers, regardless of them merely being allowed in the container, and the container being restricted to links not external to those domains) — I still want to be able to open Google News in the default container as well, without logging into my Google account, or to open it in a second or third Google container, with different customizations in those Google accounts)
- the ability to have fine-grained control over where _**external**_ links clicked in that container will _**open**_, whether in the default container ("no container"), a private window without any container, or even possibly a custom container (temporary or persisted) ... to work in concert with whatever cookie deleting scheme I happen to be using (say, the Cookie Quick Manager addon, or other Firefox settings directly).

<hr>

### Slightly longer version

OMG yes! This is the root of my own separate issue too and absolutely needs to be a feature!!!

I want to be able to have two separate containers for _two different Google accounts_ (e.g., Google1 and Google2 containers). I only have two accounts to have two different Gmail accounts, otherwise I just use the first account for everything other than email, i.e., for Calendar, Drive, YouTube, Google News). Now I create two separate containers, and so far so good.

The problem is when I open Google News in one of these containers, so I can take advantage of having set dark mode for Google News, and not seeing the new sources I have blocked, now anytime I want to click and read a news article I have remember to manually take an extra step or two and right-click to _open the external article link somewhere outside of this container,_ so that this container is not polluted with non-Google cookies).

So intuitively I think I will just try toggling on the option to _limit this container to Google domains_ .... well that doesn't work because I haven't also configured Google domains to "always open in" this container. Why? Because I have multiple Google accounts, so I literally can't have Google always just open in _one_ of my Google containers!

What I really need is an additional option to choose which domains are _**allowed**_ in any given container, and then _restricting_ that container to only those domains (and it may be the exact same allow-then-restrict list of domains in all of my Google containers, or Facebook containers, or Twiiter containers and so forth ... I always want links to external domains to open in the default container so I can later clear that "context" of cookies and local storage with the Cookie Quick Manager addon .... or even better, have those external links always open in a private window so I never have to manually go clear the default container context. The only cookies and local storage I want preserved is for Google, Facebook, Twitter, etc. in their own containers, possibly multiple containers for multiple accounts.

**Comment by dannycolin** [comment]
Currently, it isn't possible to allow the same domain name in more than one container if "Limit to designated sites" is enabled. There are a few older issues asking for the same feature of allowing a domain to be assigned to more than one container (e.g. https://github.com/mozilla/multi-account-containers/issues/1887). I'll close this one to avoid duplicates.
